VAULT-36949: Automated snapshot cloud permissions (#1145)
hashicorp/web-unified-docs: content/vault/v1.21.x (rc)/content/api-docs/system/storage/raftautosnapshots.mdx

- `aws_s3_region` `(string: <required>)` - AWS region bucket is in.

> - `aws_access_key_id` `(string)` - AWS access key ID. Make sure that this account has the following permissions in S3, so that Vault can store and manage the snapshots:

Reviewer: ```suggestion
- `aws_access_key_id` `(string)` - AWS access key ID. The assocaited AWS account
  must have the following permissions in S3 so that Vault can store and manage
  the snapshots:
```

  - `s3:ListBucket`
  - `s3:PutObject`
  - `s3:DeleteObject`

>   If you enable snapshot autoloading or want to manually load a snapshot from this configuration, the account also needs:

Reviewer: ```suggestion
  If you want to use the configration to enable snapshot autoloading or to
  manually load a snapshot, the account also needs:
```

- `aws_s3_force_path_style` `(boolean)` - Use the endpoint/bucket URL style
  instead of bucket.endpoint. May be needed when setting `aws_s3_endpoint`.

> - `aws_s3_enable_kms` `(boolean)` - Use KMS to encrypt bucket contents. If this is true, make sure that your AWS account also has the permissions:

Reviewer: ```suggestion
- `aws_s3_enable_kms` `(boolean)` - Use KMS to encrypt bucket contents. To use
  KMS encryption, the associated AWS account must have the following permissions:
```

  `google_endpoint`.

#### storage_type=azure-blob
 
- `azure_container_name` `(string: <required>)` - Azure container name to write
  snapshots to.

- `azure_account_name` `(string)` - Azure account name. Make sure that this account has

>   the following permissions in Azure Blob Storage, so that Vault can store and manage the snapshots:

Reviewer: ```suggestion
- `azure_account_name` `(string)` - Azure account name. The associated Azure
  account must have the following permissions in Azure Blob Storage so that
  Vault can store and manage the snapshots:
```
